smoke/host/util.go

```diff
@@ -0,0 +1,295 @@
+package host
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/evergreen-ci/evergreen"
+	"github.com/evergreen-ci/evergreen/agent"
+	"github.com/evergreen-ci/evergreen/rest/model"
+	"github.com/evergreen-ci/evergreen/smoke/internal"
+	"github.com/evergreen-ci/utility"
+	"github.com/mongodb/grip"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/require"
+)
+
+// SmokeTestParams contain common parameters used by smoke tests.
+type SmokeTestParams struct {
+	internal.APIParams
+	ExecModeID     string
+	ExecModeSecret string
+	CLIConfigPath  string
+	ProjectID      string
+	BVName         string
+}
+
+// GetSmokeTestParamsFromEnv gets the parameters for the host smoke test from
+// the environment. It sets defaults where possible. Note that the default data
+// depends on the setup test data for the smoke test.
+func GetSmokeTestParamsFromEnv(t *testing.T) SmokeTestParams {
```

Comment:
Is it really worth having multiple packages in this smoke directory? I'd advocate for flattening it -- then we can keep everything private, and I think it's a bit clearer that everything in here exists only to be used here. This package isn't going to grow to the extent that I think we need. (I could see a case for keeping host and container separate but it seems like that could just handled with file names)

Reply:
The last bit works for me, thanks!